src/irmin-pack/pack_store.ml

```diff
@@ -147,9 +167,109 @@ module Maker (Index : Pack_index.S) (K : Irmin.Hash.S with type t = Index.key) :
       assert (n = K.hash_size);
       decode_bin_hash (Bytes.unsafe_to_string buf) (ref 0)
 
+    type span = { offset : int63; length : int }
+    (** The type of contiguous ranges of bytes in the pack file. *)
+
+    (** Refer to the index for the position of a pack entry, assuming it is
+        indexed: *)
+    let get_entry_span_from_index_exn t hash : span =
+      match index_direct t hash with
+      | Some key' -> (
+          match Pack_key.inspect key' with
+          | Direct { offset; length; _ } -> { offset; length }
+          | Indexed _ ->
+              (* [index_direct] returns only [Direct] keys. *)
+              assert false)
+      | None ->
+          corrupted_store "Unexpected object %a missing from index" pp_hash hash
+
+    module Entry_prefix = struct
+      type t = {
+        hash : hash;
+        kind : Pack_value.Kind.t;
+        value_length : int option;
+            (** Length of the value segment including the length header (if it
+                exists). *)
```

Comment:
Thanks. I've added this to the documentation in `pack_value.mli` and referenced that from here (and also renamed the field to `size_of_value_and_length_header` to try to avoid being needlessly confusing :stuck_out_tongue:).